packages/common/container-definitions/src/loader.ts

```diff
@@ -102,23 +102,134 @@ export interface ICodeAllowList {
 }
 
 /**
- * Events emitted by the Container "upwards" to the Loader and Host
+ * Events emitted by the {@link IContainer} "upwards" to the Loader and Host.
  */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IContainerEvents extends IEvent {
+    /**
+     * Emitted when the readonly state of the container changes.
+     *
+     * @remarks Listener parameters:
+     *
+     * - `readonly`: Whether or not the container is now in a readonly state.
+     *
+     * @see {@link IContainer.readOnlyInfo}
+     */
     (event: "readonly", listener: (readonly: boolean) => void): void;
+
+    /**
+     * Emitted when the {@link IContainer} completes connecting to the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.connect}
+     */
     (event: "connected", listener: (clientId: string) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     *
+     * - `proposal`: TODO
+     *
+     * @see {@link IContainer.proposeCodeDetails}
+     */
     (event: "codeDetailsProposed", listener: (codeDetails: IFluidCodeDetails, proposal: ISequencedProposal) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     */
     (event: "contextChanged", listener: (codeDetails: IFluidCodeDetails) => void);
-    (event: "disconnected" | "attached", listener: () => void);
+
+    /**
+     * Emitted when the {@link IContainer} becomes disconnected from the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.disconnect}
+     */
+    (event: "disconnected", listener: () => void);
+
+    /**
+     * TODO
```

Comment:
This event gets emitted when detached (new) container is attached, as represented by AttachState:

"In attached state, the data has completed upload to the service.  It can be accessed by other clients after reaching attached state."

Interesting question here is about usability. In most of the case scenario caller would simply wait on async `attach` call to return.  However, if they created large, detached document, they may want to async wait on completion of this call. One alternative is to simply wait on attach call promise to resolve, the other one is to wait on this event. It seems like having this event adds some value, but there are alternatives.

Reply:
I was wondering about this as well. The event does not seem to have any usages within our repo. We have tests for it, so I believe it works correctly, but it does make me wonder if we need this.

Reply:
Interested if others have any thoughts on this. Should we mark it as deprecated in favor of simply awaiting the attach method call?

Reply:
I generally consider it good practice to consistently have an event for any state that may transition, so I'd probably say keep it even though it's a less common scenario.